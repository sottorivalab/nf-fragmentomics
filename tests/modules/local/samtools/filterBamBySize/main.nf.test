nextflow_process {

    name "Test Process FILTERBAMBYSIZE"
    script "modules/local/samtools/filterBamBySize/main.nf"
    process "FILTERBAMBYSIZE"

    test("Should run without failures") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = [
                    [ caseid:'test', sampleid: 'test', timepoint: 'T1' ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.filtered, process.out.versions).match()
        }

    }

}
