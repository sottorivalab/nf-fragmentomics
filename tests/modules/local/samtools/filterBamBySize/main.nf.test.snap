{
    "Should run without failures": {
        "content": [
            [
                [
                    {
                        "caseid": "test",
                        "sampleid": "test",
                        "timepoint": "T1"
                    },
                    "test.paired_end.sorted.filtered.bam:md5,d3565dd09b19fcb726f0bf52a7389cbd",
                    "test.paired_end.sorted.filtered.bam.bai:md5,65d3480cba8732dc1669f2fc67301da7"
                ]
            ],
            [
                "versions.yml:md5,8e1da23404c22cba5eb736cc23529841"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-16T11:56:49.091666"
    }
}